src/devices/Mcp3xxx/Mcp33xx.cs

```diff
@@ -0,0 +1,75 @@
+
+        /// <summary>
+        /// Reads a  value from the device using pseudo-differential inputs
+        /// </summary>
+        /// <param name="valueChannel">Channel which represents the signal (valid values: 0 to channelcount - 1).</param>
+        /// <param name="referenceChannel">Channel which represents the signal ground (valid values: 0 to channelcount - 1).</param>
+        /// <returns>A value corresponding to relative voltage level on specified device channels</returns>
+        public override int ReadPseudoDifferential(int valueChannel, int referenceChannel)
+        {
+            throw new NotSupportedException("Mcp33xx device does not support ReadPseudoDifferential.");
+        }
+
+        /// <summary>
+        /// Reads a 13 bit signed value from the device using differential inputs
+        /// </summary>
+        /// <remarks>
+        /// The value that is read respresents the difference between the voltage on the value channel and the voltage on the reference channel (valueChannel Reading - referenceChannel Reading).
+        /// If the valueChannel and the referenceChannel are part of the same channel pairing then the ADC converter will internally subtract the two values. If not then the subtraction is
+        /// performed in software which may mean that errors are introduced with rapidly changing signals.
+        /// </remarks>
+        /// <param name="valueChannel">Channel which represents the signal driving the value in a positive direction (valid values: 0 to channelcount - 1).</param>
+        /// <param name="referenceChannel">Channel which represents the signal driving the value in a negative direction (valid values: 0 to channelcount - 1).</param>
+        /// <returns>A 13 bit signed value corresponding to relative voltage level on specified device channels</returns>
+        public override int ReadDifferential(int valueChannel, int referenceChannel)
+        {
+            int retval;
+
+            CheckChannelRange(valueChannel, ChannelCount);
+            CheckChannelRange(referenceChannel, ChannelCount);
+
+            if (valueChannel == referenceChannel)
+            {
+                throw new ArgumentException($"ADC differential channels must be different.", nameof(valueChannel) + " " + nameof(referenceChannel));
+            }
+
+            // check if it is possible to use hardware differential because both input channels are in the same differential channel pairing
+            if (valueChannel / 2 == referenceChannel / 2)
+            {
+                // read a value from the ADC where the channel is the channel pairing
+                retval = ReadInternal(channel: valueChannel / 2, valueChannel > referenceChannel ? InputType.InvertedDifferential : InputType.Differential, adcResolutionBits: 13);
+
+                //convert 13 bit signed to 32 bit signed
+                retval = (retval >> 12) == 0 ? retval : (int)(0xFFFFE000 | retval);
```

Comment:
would it make sense to add some utility for this? (i.e. ReadSignedInternal or just a static method to convert) - also same comment - might be better to subtract instead to make it clearer

Reply:
> @Frankenslag this mostly looks good to me. There are couple of typos and some minor things but in general ready to merge after fixing those.

I have made the changes but before updating the PR I want to do some more testing with hardware...